frontend/src/components/dashboard/Analyzer.state.ts

```diff
@@ -177,8 +235,21 @@ export const analyzerStateReducer = (
       return { ...state, selectedRightSolution: action.solutionId };
     case AnalyzerStateActionType.setSelectedSolution:
       return { ...state, selectedSolutionId: action.selectedSolutionId };
-    case AnalyzerStateActionType.setBookmarkedSolutions:
-      return { ...state, bookmarkedSolutionIds: action.bookmarkedSolutionIds };
+    case AnalyzerStateActionType.addBookmarkedSolution:
+      return {
+        ...state,
+        bookmarkedSolutionIds: [
+          ...state.bookmarkedSolutionIds,
+          action.solutionId,
```

Comment:
This can append multiple times the same ID. We should probably filter pre-existing IDs.